client/van_router_create.go

```diff
@@ -47,16 +71,40 @@ func GetVanControllerSpec(options types.VanRouterCreateOptions, van *types.VanRo
 	envVars = append(envVars, corev1.EnvVar{Name: "OWNER_NAME", Value: transport.ObjectMeta.Name})
 	envVars = append(envVars, corev1.EnvVar{Name: "OWNER_UID", Value: string(transport.ObjectMeta.UID)})
 
-	volumes := &[]corev1.Volume{}
-	mounts := &[]corev1.VolumeMount{}
+	sidecars := []*corev1.Container{}
+	volumes := []corev1.Volume{}
+	mounts := make([][]corev1.VolumeMount, 0)
+	mounts = append(mounts, []corev1.VolumeMount{})
+
+	if options.AuthMode == string(types.ConsoleAuthModeOpenshift) {
+		sidecars = append(sidecars, OauthProxyContainer("skupper-proxy-controller"))
+		envVars = append(envVars, corev1.EnvVar{Name: "METRICS_PORT", Value: "8888"})
+		envVars = append(envVars, corev1.EnvVar{Name: "METRICS_HOST", Value: "localhost"})
```

Comment:
I wonder if it makes more sense to separate out the env vars by container?

Reply:
Good suggestion, I will revise.

Reply:
It turns out the env vars are already separated by container. If this needs to be done more generally, will address in a future patch set.

Reply:
Ah, looking at it again, the env vars are all for the main container. The oauthproxy container doesn't have any env vars. Confused myself when reading. Ignore this comment.